### DISCUSSION TITLE
New checks provided by Commonality

### DISCUSSION THREAD

**Discussion by alec-chernicki** [commonalityco/commonality#209] [Ideas]
I've created some foundational checks that re-enforce best practices with the `commonality-checks-recommended` package. However, there may be more checks that developers immediately reach for or are a common requirement like `hasFile()`. What are some common foundational-level checks that you'd like to see us directly create and support?

**Comment by IKatsuba** [comment]
I have a small list:
- `hasDependencies` to check dependencies and their versions
- `banDependencies` to ban some dependencies
- `fileContent` to check the content of the file

**Comment by alec-chernicki** [comment]
Hey @IKatsuba I just wanted to let you know that I've added [`hasJsonFile`](https://www.commonality.co/docs/commonality-checks-recommended#has-json-file) and [`hasTextFile`](https://www.commonality.co/docs/commonality-checks-recommended#has-text-file) to the latest version of [`commonality-checks-recommended`](https://www.commonality.co/docs/commonality-checks-recommended). These checks allow you to see if a file exists and can check for matching content. I've added some additional examples that you can check out [here](https://www.commonality.co/docs/checks/recipes).

For the use cases you mentioned here are some examples.

Check for dependencies:
```ts
export default defineConfig({
  checks: {
    ui: [
      recommended.hasJsonFile('package.json', {
        dependencies: {
			react: "^18.0.0"
        },
      }),
    ],
  },
});
```

Check for content of a file:
```ts
export default defineConfig({
  checks: {
    "*": [
      recommended.hasTextFile('.npmignore', ["dist"]),
    ],
  },
});
```

We don't yet support negation of checks but would love to hear your use case!

**Reply by RafalFilipek** [reply]
This is very useful. It would be nice to be able to pass regular expressions as a second argument.
